tests/integ/test_multithreading.py

```diff
@@ -633,6 +641,109 @@ def change_config_value(session_):
     )
 
 
+@pytest.mark.xfail(
+    "config.getoption('local_testing_mode', default=False)",
+    reason="local testing does not execute sql queries",
+    run=False,
+)
+def test_temp_name_placeholder_for_sync(threadsafe_session):
+    from snowflake.snowpark._internal.analyzer import analyzer
```

Comment:
I just recall, do we have any multi-threading tests with the new query compilation stage?

Reply:
we do not at the moment

Reply:
@sfc-gh-aalam can you get a ticket to add the tests? we should make sure all things are tested out

Reply:
https://snowflakecomputing.atlassian.net/browse/SNOW-1758742